posts/tools/test-frame-introduction.md

```diff
@@ -0,0 +1,174 @@
+method.
+This removes the need for "manual" clean-up after each test case and overall management of the resources.
+
+Besides that, Test-Frame provides classes for easier handling of basic Kubernetes resources, such as Secret, Service, NetworkPolicy, and
+many more.
+It also has built-in `KubeClient`, which creates a basic encapsulation around the Fabric8's client, together with the possibility to
+specify `KUBE_CONTEXT` for the cluster where the tests should be executed.
+
+The library is still under development, we are planning to add more functionality like a log collector (which will collect also YAMLs of resources, Pod description, and more)
+or metrics collector, for easier gathering metrics from the pre-defined endpoints.
+
+### How I can use it?
+
+During the development, the Test-Frame library is being pushed to GitHub's Maven repository.
+So in case you would like to try the Test-Frame in your test suite before it is released, you can do that by including
+the GitHub's Maven repository and the Test-Frame dependency in your `pom.xml`:
+
+```xml
+    <repositories>
+        <repository>
+            <id>test-frame</id>
+            <name>GitHub Apache Maven Packages</name>
+            <url>https://maven.pkg.github.com/skodjob/test-frame</url>
+        </repository>
+    </repositories>
+    <dependencies>
+        <dependency>
+            <groupId>io.skodjob</groupId>
+            <artifactId>test-frame-common</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+```
+where the `test-frame-common` module contains the core functionality - like the `KubeResourceManager`.
+Additionally, there are more modules that you can use and explore in our [GitHub repository](https://github.com/skodjob/test-frame).
+
+Because we are pulling a dependency from the GitHub Maven repository, we need to provide an additional configuration in the `settings.xml`:
+
+```xml
+<settings>
+    <servers>
+        <server>
+            <id>test-frame</id>
+            <username>x-access-token</username>
+            <password>${env.GITHUB_TOKEN}</password>
+        </server>
+    </servers>
+</settings>
+```
+
+Where the `id` of the server has to match with the `id` of the repository in the `pom.xml`.
+
+Once you are all set, you can try to experiment with the library in the tests by annotating the test classes with
+`@ResourceManager` annotation, determining that the `KubeResourceManager` should manage the resources created in the tests.
+
+```java
+@ResourceManager
+class NamespaceCreationST {
+    
+    @Test
+    void testThatNamespaceExists() {
+        KubeResourceManager.getInstance().createResourceWithWait(
+                new NamespaceBuilder().withNewMetadata().withName("test").endMetadata().build());
+        assertNotNull(KubeResourceManager.getKubeCmdClient().get("namespace", "test"));
+    }
+}
```

Comment:
Should we add a more comprehensive example?

Reply:
I thought about it for a while, but this should be just intro. I added there link to the GH repo, where are more examples. Maybe we should write more than just README (so maybe small docs), which will contain more of the examples.